homeworks/rostyslav.havuka_RostyslavHavuka/12-httpclient/user-card/src/app/interceptors/loader.interceptor.ts

```diff
@@ -0,0 +1,16 @@
+import { Injectable } from "@angular/core";
+import { HttpEvent, HttpHandler, HttpInterceptor, HttpRequest } from "@angular/common/http";
+import { fromEvent, Observable } from "rxjs";
+import { delay, finalize } from "rxjs/operators";
+import { LoaderService } from '../services/loader.service';
+@Injectable()
+export class LoaderInterceptor implements HttpInterceptor {
+    constructor(public loaderService: LoaderService) { }
+    intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+        this.loaderService.show();
+        return next.handle(req).pipe(
+            delay(1000),
```

Comment:
Why do you set the additional delay for requests?
The user will see all loaded info only after this delay. As I see, you need to remove this delay